### PR TITLE
Action docs update

### PR DESCRIPTION
**Pull request by holdenweb**
**Please review the following checklist.**

- [x] Docstrings on all new or modified functions / classes 
- [x] Updated documentation
- [x] Updated CHANGELOG.md (where appropriate)

This is a fairly straightforward rewrite of the actions documentation, which clarifies a few issues that caused me problems.
It doesn't seem significant enough to merit a CHANGELOG entry.

### CODE REVIEW

File: docs/guide/actions.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Actions are allow-listed functions with a string syntax you can embed in links a
 
 ## Action methods
 
-Action methods are methods on your app or widgets prefixed with `action_`. Aside from the prefix these are regular methods which you could call directly if you wished.
+Actions are handled by _action methods_ of your app or widgets - methods whose names are prefixed with `action_`. These are regular methods which you can call directly, but normally they are triggered by actions.
```
Comment (holdenweb) [inline docs/guide/actions.md:7]: This change is partly pedantry and partly a wish to add more context, but "aside from the prefix" seems completely redundant. It's quite probably a matter of taste and style preference.

File: docs/guide/actions.md [hunk 2]
```diff
@@ -16,9 +16,9 @@ Let's write an app with a simple action.
 --8<-- "docs/examples/guide/actions/actions01.py"
 ```
 
-The `action_set_background` method is an action which sets the background of the screen. The key handler above will call this action if you press the ++r++ key.
+The `action_set_background` method is an action method that sets the background of the screen. The key handler above will call this action if you press the ++r++ key.
```
Comment (holdenweb) [inline docs/guide/actions.md:19]: A method isn't an action, and sloppy usage will inevitably lead to misunderstanding on the part of at least some readers.

File: docs/guide/actions.md [hunk 3]
```diff
@@ -16,9 +16,9 @@ Let's write an app with a simple action.
...
 
-Although it is possible (and occasionally useful) to call action methods in this way, they are intended to be parsed from an _action string_. For instance, the string `"set_background('red')"` is an action string which would call `self.action_set_background('red')`.
+Although it is possible (and occasionally useful) to call action methods directly in this way, they are more often parsed from an _action string_. For instance, the string `"set_background('red')"` is an action string which would call `action_set_background('red')`.
```
Comment (holdenweb) [inline docs/guide/actions.md:21]: A re-reading convinces me this change was based on my misreading without the context of visible code, so I'll modify the PR to exclude it.

File: docs/guide/actions.md [hunk 4]
```diff
@@ -32,25 +32,32 @@ You will not typically need this in a real app as Textual will run actions in li
...
 
-Action strings have the following format:
+Action strings therefore have the following format:
 
-- The name of an action on is own will call the action method with no parameters. For example, an action string of `"bell"` will call `action_bell()`.
-- Actions may be followed by braces containing Python objects. For example, the action string `set_background("red")` will call `action_set_background("red")`.
-- Actions may be prefixed with a _namespace_ (see below) followed by a dot.
+- The name of an action (with or without a namespace) on its own will call the action method with no parameters in the given namespace, if any. For example, an action string of `"app.bell"` will call your app's `action_bell()` method.
+- The action name may be followed by parentheses optionally containing a list of Python objects, which become parameters to the action method call. For example, the action string `set_background("red")` will first try to call `action_set_background("red")` as a widget method, then as a screen method, and finally as an app method.
```
Comment (holdenweb) [inline docs/guide/actions.md:52]: This is simply a refactoring of the explanation. The first bullet explains parameterless actions while pointing out this is orthogonal to the presence of a namespace. The second demonstrates actions with parameters and explains the lookup sequence when no namespace is present. "Braces" was, of course, just plain wrong, so I presume it's a holdover from an earlier design.

File: docs/guide/actions.md [hunk 5]
```diff
@@ -32,25 +32,32 @@ You will not typically need this in a real app as Textual will run actions in li
...
 
 <div class="excalidraw">
 --8<-- "docs/images/actions/format.excalidraw.svg"
 </div>
 
 ### Parameters
 
-If the action string contains parameters, these must be valid Python literals. Which means you can include numbers, strings, dicts, lists etc. but you can't include variables or references to any other Python symbols.
+If the action string contains parameters, these must be valid Python literals. So you can include numbers, strings, dicts, lists etc. but NOT variables or references to any other Python symbols.
```
Comment (holdenweb) [inline docs/guide/actions.md:60]: A simplifying rewording, I felt. Again, perhaps largely a matter of opinion.

**Review by holdenweb — COMMENTED** [review]
The comments were'nt meant to be [art of a formal review, but it's easier to log them this way than re-create the comments outside a review.

**Comment by holdenweb** [comment]
Thanks for your comment, Rodrigo.

I did indeed have issues with the documentation, though clearly I only have my own personal experience to go on.

Having learned about events and event-handling, and being somewhat familiar with DOM concepts, I found that the different handling sequence for actions was insufficiently well-explained.  In reviewing documentation I try to maintain the viewpoint of someone coming to the software for the first time, which isn't that far wrong in textual's case.

I've added some comments to the changes, and will revert one change immediately.

**Comment by rodrigogiraoserrao** [comment]
Non-trivial documentation changes are inglorious contributions to be made by external contributors because it is difficult to balance the voice of the documentation, which we want to be consistent, together with the information layout and presentation that suits best each reader.

I've opened #3511 as a follow-up to this PR, making use of some of the more objective improvements that you made with this PR and I've added you as a co-author there.
In addition to that, I will close this PR.

In the future, we encourage you to open (discussion) issues pointing out deficiencies in the documentation so that we can figure out the best way to fix those.
This should help minimise the time you invest in making the changes yourself only to figure out that your personal preferences clash with mine or someone else's.
We'll also change the contributing guide to add information to this end so that others in the future are less likely to hit the same roadblock as you did.

Thanks for your understanding 🙏

**Comment by rodrigogiraoserrao** [comment]
Sorry about that. We have a 🇵🇹  phrase that I tried to translate without success!
I've reworded the “inglorious” bit.

**Comment by holdenweb** [comment]
Thanks very much for the feedback. I'll engage more closely on Discord before making a PR in future.
